packages/react-avatar/src/components/Avatar/Avatar.types.ts

```diff
@@ -134,24 +135,33 @@ export type AvatarNamedColor =
   | 'anchor';
 
 /**
- * Names of the shorthand properties in AvatarProps
+ * Properties for Avatar
  */
-export type AvatarShorthandPropsCompat = 'label' | 'image' | 'badge' | 'icon';
+export interface AvatarProps extends ComponentProps<AvatarSlots>, Partial<AvatarCommons> {
+  /**
+   * Badge to show the avatar's presence status.
+   * Can either be a string indicating the status ("busy", "away", etc.), or a PresenceBadgeProps object.
+   */
+  badge?: PresenceBadgeStatus | ObjectShorthandProps<PresenceBadgeProps>;
```

Comment:
```suggestion
  badge?: PresenceBadgeStatus | ShorthandProps<PresenceBadgeProps>;
```

@behowell I can understand the case with `image`, but can you please clarify why we should not use `ShorthandProps` there? With `ObjectShorthandProps` it will be impossible to pass JSX elements:

```tsx
<Avatar badge={<span />} />
```

Reply:
the `ComponentProps` type actually exposes the props as `ShorthandProps<ObjectShorthandProps>`